### PR TITLE
adding pT-dependent DCA cut

### PR DESCRIPTION
**Pull request by glromane**

### CODE REVIEW

**Comment by glromane** [comment]
Dear @victor-gonzalez ,

In this commit we introduce a possibility to use a pT-dependent cut on DCA. Could you please approve it. Thank you.

Sincerely yours, Gleb.

@njacazio @sofiatomassini

File: PWGCF/Femto3D/Tasks/femto3dPairTask.cxx [hunk 1]
```diff
@@ -389,6 +386,8 @@ struct FemtoCorrelations {
         continue;
       if (track.template singleCollSel_as<soa::Filtered<FilteredCollisions>>().hadronicRate() < _IRcut.value.first || track.template singleCollSel_as<soa::Filtered<FilteredCollisions>>().hadronicRate() >= _IRcut.value.second)
         continue;
+      if (abs(track.dcaXY()) > _dcaXY.value[0] + _dcaXY.value[1] * std::pow(track.pt(), _dcaXY.value[2]) || abs(track.dcaZ()) > _dcaZ.value[0] + _dcaZ.value[1] * std::pow(track.pt(), _dcaZ.value[2]))
```
Comment (victor-gonzalez) [inline PWGCF/Femto3D/Tasks/femto3dPairTask.cxx:389]: Have you considered if using two `TF1` would be more efficient?

**Review by victor-gonzalez — APPROVED** [review]

File: PWGCF/Femto3D/Tasks/femto3dPairTask.cxx (hunk 1 above)
Comment (glromane) [reply PWGCF/Femto3D/Tasks/femto3dPairTask.cxx:389]: Dear @victor-gonzalez ,

thanks for a nice suggestion! I haven't thought of it. I did a quick test for the exec time with two methods:
1) set up a TF1 and evaluate for a given "x";
2) calculate straightforward; -- appears to be faster

Below I attach the code that I used (if you meant to use another method of TF1, please tell me, I'll check it out). in the comments to 2 used methods I put exec. times. Thank you.


#include "TF1.h"
#include "TRandom.h"
#include "TStopwatch.h"

void test() {
  float par1 = 0.004f;
  float par2 = 0.013f;
  float par3 = -1.0f;

  TF1 fff = TF1("fff", "[0]+[1]*pow(x, [2])");
  fff.SetParameter(0, par1);
  fff.SetParameter(1, par2);
  fff.SetParameter(2, par3);
  TStopwatch t;
  t.Start();
  // your code goes here

  for (int i = 0; i < 100000000; i++) {
    //fff.Eval(gRandom->Uniform(0, 10)); // calculate with TF1 => Real time 0:00:02, CP time 2.800
    par1+par2*pow(gRandom->Uniform(0, 10), par3); // straightforward => Real time 0:00:01, CP time 1.710
  }
  t.Stop();
  t.Print();
}

Comment (victor-gonzalez) [reply PWGCF/Femto3D/Tasks/femto3dPairTask.cxx:389]: Interesting!
Thanks for checking it!

**Pull request merged**